### PR TITLE
Add e2e tests for redis optimized and none strategy

### PR DESCRIPTION
**Pull request by emmaz90**
# JS SDK

## What did you accomplish?
Add e2e tests for redis optimized and none strategy

## Extra Notes

### CODE REVIEW

File: src/__tests__/consumer/node_redis.spec.js [hunk 1]
```diff
@@ -146,6 +147,183 @@ tape('NodeJS Redis', function (t) {
...
+        assert.equal(typeof client.track().then, 'function', 'Track calls should always return a promise on Redis mode, even when parameters are incorrect.');
+
+        assert.true(await client.track('nicolas@split.io', 'user', 'test.redis.event', 18), 'If the event was succesfully queued the promise will resolve to true');
+        assert.false(await client.track(), 'If the event was NOT succesfully queued the promise will resolve to false');
+
+        await client.ready(); // promise already resolved
+        await client.destroy();
+
+        // Validate stored impressions and events
+        exec(`echo "LLEN ${config.storage.prefix}.SPLITIO.impressions \n LLEN ${config.storage.prefix}.SPLITIO.events" | redis-cli  -p ${redisPort}`, (error, stdout) => {
```
Comment (EmilianoSanchez) [inline src/__tests__/consumer/node_redis.spec.js:224]: We should check that the Redis value `${config.storage.prefix}.SPLITIO.impressions.count` is the expected

File: src/__tests__/consumer/node_redis.spec.js [hunk 2]
```diff
@@ -146,6 +147,183 @@ tape('NodeJS Redis', function (t) {
...
+        });
+      });
+  });
+  
+  t.test('Regular usage - NONE strategy', assert => {
+    config.sync.impressionsMode = NONE;
+    initializeRedisServer()
+      .then(async (server) => {
+        const expectedConfig = '{"color":"brown"}';
+        assert.equal(config.sync.impressionsMode, NONE, 'impressionsMode should be OPTIMIZED');
```
Comment (EmilianoSanchez) [inline src/__tests__/consumer/node_redis.spec.js:252]: `impressionsMode should be NONE`

File: src/__tests__/consumer/node_redis.spec.js [hunk 3]
```diff
@@ -146,6 +147,183 @@ tape('NodeJS Redis', function (t) {
...
+
+        // Validate stored impressions and events
+        exec(`echo "LLEN ${config.storage.prefix}.SPLITIO.impressions \n LLEN ${config.storage.prefix}.SPLITIO.events" | redis-cli  -p ${redisPort}`, (error, stdout) => {
+          if (error) assert.fail('Redis server should be reachable');
+
+          const trackedImpressionsAndEvents = stdout.split('\n').filter(line => line !== '').map(line => parseInt(line));
+          assert.deepEqual(trackedImpressionsAndEvents, [0, 2], 'Tracked impressions and events should not be stored in Redis');
+
+          // Validate stored telemetry
+          exec(`echo "HLEN ${config.storage.prefix}.SPLITIO.telemetry.latencies \n HLEN ${config.storage.prefix}.SPLITIO.telemetry.exceptions \n HGET ${config.storage.prefix}.SPLITIO.telemetry.init nodejs-${version}/${HOSTNAME_VALUE}/${IP_VALUE}" | redis-cli  -p ${redisPort}`, (error, stdout) => {
```
Comment (EmilianoSanchez) [inline src/__tests__/consumer/node_redis.spec.js:309]: We should check that the Redis value `${config.storage.prefix}.SPLITIO.uniquekeys` is the expected
